client-app/src/Components/DonorForm.tsx

```diff
@@ -52,11 +64,15 @@ const DonorForm: React.FC = () => {
     const newErrors: FormErrors = {};
     if (!formData.firstName.trim()) newErrors.firstName = 'First Name is required';
     if (!formData.lastName.trim()) newErrors.lastName = 'Last Name is required';
+    if (!formData.contact.trim()) newErrors.contact = 'Contact is required';
     if (!formData.email.trim()) {
       newErrors.email = 'Email is required';
     } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) {
       newErrors.email = 'Invalid email format';
     }
+    if (!formData.addressLine1.trim()) newErrors.addressLine1 = 'Address Line 1 is required';
+    if (!formData.state.trim()) newErrors.state = 'State is required';
+    if (!formData.city.trim()) newErrors.city = 'City is required';
     if (!formData.zipcode.trim()) {
```

Comment:
This feels like a lot of similar if statements in a row. Is there a way to condense this or maybe create a function for these errors?

Reply:
That's a great suggestion. Feel free to use AI to figure out how to improve code segments like this. Here's an AI suggested generalization (it could probably benefit from some comments). Overall, the code is longer but it's more generic and reusable. If you have other forms in this project, the validation function could be reused there.

```
const validateField = (value, rule) => {
  if (typeof rule === 'function') {
    return rule(value);
  }
  if (rule === 'required' && !value.trim()) {
    return `${value} is required`;
  }
  return null;
};

const validateForm = (formData, validationRules) => {
  const newErrors = {};
  
  Object.entries(validationRules).forEach(([field, rule]) => {
    const error = validateField(formData[field], rule);
    if (error) {
      newErrors[field] = error;
    }
  });

  return newErrors;
};

const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

const validationRules = {
  firstName: 'required',
  lastName: 'required',
  contact: 'required',
  email: (value) => {
    if (!value.trim()) return 'Email is required';
    if (!emailRegex.test(value)) return 'Invalid email format';
    return null;
  },
  addressLine1: 'required',
  state: 'required',
  city: 'required'
};

const newErrors = validateForm(formData, validationRules);
```